cmd/migration/v0.9.1-v1/mapper/vault_wallet.go

```diff
@@ -0,0 +1,118 @@
+package mapper
+
+import (
+	"fmt"
+
+	v091domain "github.com/tdex-network/tdex-daemon/cmd/migration/v0.9.1-v1/v091-domain"
+	v1domain "github.com/tdex-network/tdex-daemon/cmd/migration/v0.9.1-v1/v1-domain"
+)
+
+const (
+	FeeAccount = iota
+	WalletAccount
+	FeeFragmenterAccount
+	MarketFragmenterAccount
+	UnusedAccount3
+
+	nameSpaceFormat = "bip84-account%d'"
+)
+
+func (m *mapperService) FromV091VaultToV1Wallet(
+	vault v091domain.Vault,
+	walletPass string,
+) (*v1domain.Wallet, error) {
+	accounts := make(map[string]*v1domain.Account)
+	accountsByLabel := make(map[string]string)
+	highestAccountIndex := 0
+	masterKey, err := vault.MasterKey(walletPass)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, v := range vault.Accounts {
+		label, err := m.getLabel(v.AccountIndex)
+		if err != nil {
+			return nil, err
+		}
+
+		namespace := fmt.Sprintf(nameSpaceFormat, v.AccountIndex)
+		accountsByLabel[label] = namespace
+
+		if v.AccountIndex > highestAccountIndex {
+			highestAccountIndex = v.AccountIndex
+		}
+
+		xpub, err := v091domain.Xpub(
+			uint32(v.AccountIndex),
+			masterKey,
+		)
+		if err != nil {
+			return nil, err
+		}
+
+		accounts[namespace] = &v1domain.Account{
+			AccountInfo: v1domain.AccountInfo{
+				Namespace: namespace,
+				Label:     label,
+				Xpub:      xpub,
+				DerivationPath: fmt.Sprintf(
+					"%s/%d", v091domain.RootPath, v.AccountIndex,
+				),
+			},
+			Index:                  uint32(v.AccountIndex),
+			BirthdayBlock:          0,
+			NextExternalIndex:      uint(v.LastExternalIndex + 1),
+			NextInternalIndex:      uint(v.LastInternalIndex + 1),
+			DerivationPathByScript: v.DerivationPathByScript,
+		}
+	}
+
+	mnemonic, err := v091domain.Decrypt(
+		[]byte(vault.EncryptedMnemonic), []byte(walletPass),
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	//there is slight difference between how v1 and v091 wallet encrypt/decrypt
+	//the mnemonic, in order to be compatible with the v1 wallet, we need to
+	//encrypt it again same way v1 does
+	encryptedMnemonic, err := v091domain.Encrypt(string(mnemonic), walletPass)
+	if err != nil {
+		return nil, err
+	}
+
+	return &v1domain.Wallet{
+		EncryptedMnemonic:   []byte(encryptedMnemonic),
+		PasswordHash:        vault.PassphraseHash,
+		BirthdayBlockHeight: 0,
+		RootPath:            v091domain.RootPath,
+		NetworkName:         vault.Network.Name,
+		Accounts:            accounts,
+		AccountsByLabel:     accountsByLabel,
+		NextAccountIndex:    uint32(highestAccountIndex + 1),
+	}, nil
+}
+
+func (m *mapperService) getLabel(accountIndex int) (string, error) {
+	switch accountIndex {
+	case FeeAccount:
+		return "fee_account", nil
+	case WalletAccount:
+		return "wallet_account", nil
```

Comment:
Ah! I didn't think about migrating this account, which is no more used or supported in v1. Anyway, I think it's fine to call it `personal_account`, which clarifies better the purpose of this account, the user will change it later by using the ocean cli eventually:
```suggestion
	case WalletAccount:
		return "personal_account", nil
```